contracts/staking/PoSValidatorManager.sol

```diff
@@ -350,13 +375,22 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         // the complete step, even if the delivered nonce is greater than the nonce used to
         // initialize the removal.
         delegator.status = DelegatorStatus.PendingRemoved;
-        delegator.endedAt = uint64(block.timestamp);
+        delegator.endedAt = currentTime;
         delegator.endingNonce = nonce;
 
         $._delegatorStakes[delegationID] = delegator;
 
-        Validator memory validator = getValidator(validationID);
-        require(validator.weight > delegator.weight, "PoSValidatorManager: Invalid weight");
+        if (includeUptimeProof) {
```

Comment:
seems similar to that of validator end, any helpers that can be used?

Reply:
Not as far as I can think...